File: Data Structures & Algorithms/house-robber/submission-0.java
class Solution {
    public int rob(int[] nums) {
        int[] max = new int[nums.length];
        if (nums.length == 1) {
            return nums[0];
        }
        if (nums.length == 2) {
            return Math.max(nums[0], nums[1]);
        }
        max[0] = nums[0];
        max[1] = Math.max(nums[0], nums[1]);
        for (int i = 2; i < nums.length; i++) {
            max[i] = Math.max(nums[i] + max[i-2], max[i-1]);
        }
        return max[nums.length-1];
    }
}

File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        
        curr = []
        res = []
        pick = [False] * len(nums)
        
        def backtrack(curr):
            if len(curr) == len(nums):
                res.append(curr.copy())
                return
            for i in range(len(nums)):
                if not pick[i]:
                    curr.append(nums[i])
                    pick[i] = True
                    backtrack(curr)
                    curr.pop()
                    pick[i] = False

        backtrack(curr)
        

        return res